R/mutate_super_vars.R

```diff
@@ -0,0 +1,210 @@
+                                        colname_div293_tax = "div293_tax", 
+                                        colname_new_Taxable_Income = "new_Taxable_Income",
+                                        div293_threshold = 300e3, 
+                                        # for low income tax contributions amount
+                                        cap = 25e3, cap2 = 35e3, age_based_cap = TRUE, cap2_age = 59, ecc = FALSE,
+                                        use_other_contr = FALSE,
+                                        div293 = TRUE, warn_if_colnames_overwritten = TRUE, drop_helpers = FALSE, copyDT = TRUE){
+  # Todo/wontfix
+  if (!identical(ecc, FALSE)) stop("ECC not implemented.")
+  
+  # CRAN NOTE avoidance
+  age_range_description <- concessional_cap <- div293_income <- div293_tax <-
+    low_tax_contributions_div293 <- pre_tax_contributions <- rental_losses <-
+    super_income_in_excess_of_cap <- surchargeable_income_div293 <-
+    concessional_contributions <- new_Taxable_Income <- .new_Taxable_Income <-
+    excess_concessional_contributions <- NULL
+  
+  if (copyDT){
+    .sample.file <- data.table::copy(.sample.file)
+  }
+  
+  if (!data.table::is.data.table(.sample.file)){
+    stop(".sample.file must be a data.table")
+  }
+  
+  # Ensure the user is appropriately warned if we are about to overwrite columns:
+  new_colnames <- c("concessional_cap", "div293_income", "div293_tax", 
+                    "low_tax_contributions_div293", "pre_tax_contributions", "rental_losses", 
+                    "excess_concessional_contributions", "surchargeable_income_div293", 
+                    "concessional_contributions", 
+                    ".new_Taxable_Income")
+  
+  common.indices <- new_colnames %in% names(.sample.file)
+  
+  if (warn_if_colnames_overwritten && any(common.indices)){
+    warning("Overwriting", names(.sample.file)[common.indices], "in .sample.file")
+  }
+  
+  if (colname_new_Taxable_Income == "Taxable_Income"){
+    warning("Dropping Taxable Income.")
+    orig_colname_new_Taxable_Income <- colname_new_Taxable_Income
+    colname_new_Taxable_Income <- ".Taxable_Income"
+  }
+  
+  if (any(c(colname_div293_tax, colname_concessional, colname_new_Taxable_Income) %in% names(.sample.file))){
+    warning("Dropping requested column names.")
+    .sample.file[ , which(names(.sample.file) %in% c(colname_div293_tax, colname_concessional, colname_new_Taxable_Income)) := NULL]
+  }
+  
+  if (!all(c("MCS_Emplr_Contr", 
+             "MCS_Prsnl_Contr", 
+             "MCS_Othr_Contr", 
+             "MCS_Ttl_Acnt_Bal") %in% names(.sample.file))){
+    stop("The sample file you requested does not have the variables needed for this function.")
+  }
+  
+  # Concessional contributions
+  if (use_other_contr){
+    .sample.file[ , concessional_contributions := MCS_Emplr_Contr + Non_emp_spr_amt + MCS_Othr_Contr]
+  } else {
```

Comment:
Note that by default this is not included in either. It's only if the user specifically requests it.